[RHACS] ROX-23413: Documenting how frequently new definitions are availabe for downloading (#77628)
openshift/openshift-docs: configuration/enable-offline-mode.adoc

[id="update-scanner-definitions-in-offline-mode"]
== Updating Scanner definitions in offline mode

> Scanner contains a local vulnerability definitions database. If {rh-rhacs-first} is running in normal mode with an internet connection, Scanner fetches new vulnerability definitions from the internet and updates its database.

Author: yeah, wecan do with normal mode.

//TODO: Add link to Vulnerability definitions

However, when you are using {product-title-short} in offline mode, you must manually update Scanner definitions by uploading them to Central.

> If {product-title-short} is running in offline mode, Scanner checks every 3 hours whether new definitions are available from Central. If new definitions are available, Scanner downloads the new definitions from Central, marks them as default and then uses the updated definitions to scan images.

Reviewer: > If {product-title-short} is running in offline mode, Scanner checks every 3 hours whether new definitions are available from Central.

That's not quite true. Scanner checks with Central every ~5 minutes (plus some jitter). Central checks if there is new data also every ~5 minutes + some jitter. The data is updated elsewhere ~every 3 hours (though this is very subject to change)

[id="update-scanner-definitions-in-offline-mode"]
== Updating Scanner definitions in offline mode

> Scanner maintains a database of vulnerabilities that it retrieves from Central. When {rh-rhacs-first} is running in normal mode, Central retrieves the latest vulnerability data from the internet and passes it on to Scanner.

Reviewer 2: ```suggestion
Scanner maintains a database of vulnerabilities. When {rh-rhacs-first} runs in normal mode, Central retrieves the latest vulnerability data from the internet and sends the data to Scanner.
```
better to remove personification and "it, which is not searchable or clear
The rest of this section talks about Scanner receiving the information from Central, not retrieving it. Which is it?

Reviewer 3: FWIW - Scanner 'pulls' vuln data from Central, and Central pulls the vuln data from the internet.  Central never 'initiates' sending vuln data to Scanner.  Perhaps:

```suggestion
Scanner maintains a database of vulnerabilities. When {rh-rhacs-first} runs in normal mode, Central retrieves the latest vulnerability data from the internet, and Scanner retrieves vulnerability data from Central.
```

//TODO: Add link to Vulnerability definitions

> However, if you are using {product-title-short} in offline mode, you must manually update the vulnerability data. To manually update the vulnerability data, you must upload a bundle to Central, which Scanner then retrieves.

Reviewer 3: > ...you must upload a bundle to Central

nit: I don't believe we use `bundle` elsewhere, which may be OK given the context. Otherwise perhaps something along the lies of `definitions file` or `archive containing definitions`

> However, if you are using {product-title-short} in offline mode, you must manually update the vulnerability data. To manually update the vulnerability data, you must upload a definitions file to Central, which Scanner then retrieves.

Reviewer 3: Scanner pulls the 'definitions files' which contains vulnerability data (amongst other things)

> In both online and offline mode, Scanner checks for new data from Central every 5 minutes by default. In online mode, Central also checks for new data approximately every 5 - 20 minutes.

Reviewer 2: ```suggestion
In both online and offline mode, Scanner checks for new data from Central every 5 minutes by default. In online mode, Central checks for new data from the internet approximately every 5-20 minutes.
```

[id="update-scanner-definitions-in-offline-mode"]
== Updating Scanner definitions in offline mode

> Scanner contains a local vulnerability definitions database. If {rh-rhacs-first} is running in normal mode, Scanner fetches new vulnerability definitions from Central. Central retrieves the updates from the internet, and then updates the Central database.

Reviewer: > and then updates the Central database

Central does store the data, yes, but we care more about Scanner storing the data.

```suggestion
Scanner maintains a database of vulnerabilities which it fetches from Central. If {rh-rhacs-first} is running in normal mode, Central fetches the latest vulnerability data from the internet to pass back to Scanner.
```

> Scanner contains a local vulnerability definitions database. If {rh-rhacs-first} is running in normal mode, Scanner fetches new vulnerability definitions from Central. Central retrieves the updates from the internet, and then updates the Central database.

Author: Thank you for the suggestions. However, it is beyond the scope of this Jira ticket. We will create a separate ticket to treat it as a content improvement activity.

> Scanner contains a local vulnerability definitions database. If {rh-rhacs-first} is running in normal mode, Scanner fetches new vulnerability definitions from Central. Central retrieves the updates from the internet, and then updates the Central database.

Author: https://issues.redhat.com/browse/ROX-24896

//TODO: Add link to Vulnerability definitions

> However, if you are using {product-title-short} in offline mode, you must manually update the definitions. 
> By uploading the definitions to Central, you can manually update the definitions.

Reviewer: I really don't like that we have been using "definitions" in our product, and I want to remove it.

```suggestion
However, if you are using {product-title-short} in offline mode, you must manually update the vulnerability data by uploading a bundle to Central, which Scanner will fetch.
```

> If the product is running in offline mode, Scanner checks with Central for new data approximately every 5 minutes. Central also checks for new data approximately every 5 minutes. 

Reviewer: In both online and offline mode, Scanner checks with Central every 5 minutes, by default.

In online mode, Central checks for new data every 5 - 20 minutes (can probably get away with just saying approximately 5).

> The data is updated approximately every 3 hours. If new definitions are available, Scanner downloads them from Central, marks them as default and then uses the updated definitions to scan images.

Reviewer: I think it makes more sense to talk about the frequency of the data's update cycle after we mention the user has to download the data, which is done below. At the moment, it's not very clear which data we are talking about.

> marks them as default

I don't know what this means

```suggestion
The offline data source is updated approximately every 3 hours. Once uploaded to Central, Scanner will download it and update its local vulnerability database.
```
